src/CTA.Rules.Analysis/RulesAnalysis.cs

```diff
@@ -184,7 +184,7 @@ private bool AnalyzeChildren(FileActions fileAction, UstList<UstNode> children,
                         case IdConstants.InvocationIdName:
                             {
                                 InvocationExpression invocationExpression = (InvocationExpression)child;
-                                var compareToken = new InvocationExpressionToken() { Key = invocationExpression.MethodName, Namespace = invocationExpression.Reference.Namespace, Type = invocationExpression.SemanticClassType };
+                                var compareToken = new InvocationExpressionToken() { Key = invocationExpression.SemanticOriginalDefinition, Namespace = invocationExpression.Reference.Namespace, Type = invocationExpression.SemanticClassType };
                                 _rootNodes.Invocationexpressiontokens.TryGetValue(compareToken, out var token);
                                 if (token != null)
                                 {
```

Comment:
This looks like the right thing to do, but were any rules affected by this?

Reply:
Oh yep, I had to fix all the invocation expression rules. There were quite a few for Owin and a couple in the system.Web rules file. Unfortunately after doing some checking the only way to fix this problem was to make the rule from precise which necessitated changing all the rules unfortunately.

Reply:
Got it, makes sense. If the unit tests are all still working then this should be ok